t/ui/28-keys_to_render_as_links.t

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env perl
+# Copyright  © 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+use Test::Most;
+
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use Test::Mojo;
+use Test::Warnings ':report_warnings';
+use lib "$FindBin::Bin/../lib";
+use OpenQA::Test::Case;
+#use OpenQA::Test::Database;
+use OpenQA::SeleniumTest;
+
+my $schema = OpenQA::Test::Case->new->init_data(fixtures_glob => '01-jobs.pl');
+# setup openqa.ini with job_settings_ui
+$ENV{OPENQA_CONFIG} = "t/data/03-setting-links";
+my $t   = Test::Mojo->new('OpenQA::WebAPI');
+my $app = $t->app;
+$t->app($app);
+
+my $jobs     = $schema->resultset('Jobs');
+my $job_id   = 99938;
+my $foo_path = "foo/foo.txt";
+my $uri_path = "/tests/$job_id/settings/$foo_path";
+
+my $driver = call_driver();
+unless ($driver) {
+    plan skip_all => $OpenQA::SeleniumTest::drivermissing;
+    exit(0);
+}
+my $url = 'http://localhost:' . OpenQA::SeleniumTest::get_mojoport;
+
+note 'Check job settings link';
+$t->get_ok("/tests/$job_id#settings")->status_is(200);
+note 'Navigate to job settings';
+$driver->get("/tests/$job_id#settings");
+note 'Find link associated with keys_to_render_as_links';
+is($driver->find_element_by_link($foo_path)->get_text(), "foo/foo.txt", "link is found");
+note "Make sure that no other settings are rendered as links";
+my @number_of_elem = $driver->find_element_by_xpath('//*[@id="settings_box"]//a');
+is(scalar @number_of_elem, 1, "Settings render as link only the proper one");
+note 'Navigate to the link and display context';
+$driver->find_element_by_link($foo_path)->click();
+is($driver->get_current_url(), "$url/tests/$job_id/settings/$foo_path", "Link is accessed with correct URI");
+$t->get_ok("$url/tests/$job_id/settings/$foo_path")->status_is(200)
+  ->content_like(qr|test|i, "source of the file loaded correctly");
```

Comment:
This test is a weird mix of using Selenium (the `$driver->` stuff) and `Test::Mojo` (the `$t->` stuff). It is generally fine to use both approaches within the same test file but here you start using `Test::Mojo`, then you're repeating the exact same query again with Selenium and conclude the tests again with a `Test::Mojo` call which is just repeating the previous Selenium query.

It does not look like there's JavaScript involved so you could actually just do everything within `Test::Mojo` (which usually runs faster and is less error prone than Selenium).

And like I said in the other comment: It would make sense to add this test into an existing test or at least combine the both tests you have to reduce the overhead.

Reply:
i have merged the two tests here. i keep them under `t/ui/` as i still make use of selenium. Although i removed statements which repeat the same test, so it should make more sense now. I might be wrong, but i need to verify that the page is rendered with the link, (and create coverage for the [helper](https://github.com/os-autoinst/openQA/pull/3203/files#diff-ce6d98ac784b7abc299796cba21eba43R379) I added, and i thought that it is simpler with the selenium.